<launch>
	<arg name="id" default="0"/>
	<arg name="ns" default="$(eval 'robot' + str( arg('id')) )"/>

	<arg name="fcu_url" default="/dev/ttyPixhawk:57600" />
	<arg name="gcs_url" default="udp://@127.0.0.1:30000" />
	<arg name="tgt_system" default="$(eval str( arg('id') + 1) )" />
	<arg name="tgt_component" default="1" />
	<arg name="indoor" default="False"/>
	<arg name="rbname" default="drone1"/>

  <include file="$(find vrpn_client_ros)/launch/sample.launch" if="$(arg indoor)">
		<arg name="server" default="192.168.0.101"/>
  </include>

	<node pkg="topic_tools" name="topic_relay_$(arg id)" type="relay" args="/vrpn_client_node/$(arg rbname)/pose /$(arg ns)/mavros/vision_pose/pose" if="$(arg indoor)"/>

	<include file="$(find formation)/launch/mavros_ns.launch">
		<arg name="fcu_url" value="$(arg fcu_url)" />
		<arg name="gcs_url" default="$(arg gcs_url)" />
		<arg name="tgt_system" default="$(arg tgt_system)" />
		<arg name="tgt_component" default="$(arg tgt_component)" />
		<arg name="log_output" default="screen" />
		<arg name="ns" default="$(arg ns)"/>
	</include>

	<group ns="$(arg ns)">
		<rosparam command="load" file="$(find formation)/config/params.yaml" />
		<rosparam command="load" file="$(find formation)/config/mavlink_bridge_params.yaml" />
		<param name="myID" type="int" value="$(arg id)" />

		<node pkg="formation" type="robot_node.py" name="robot_node_$(arg id)"  required="true" output="screen">
        </node>

        <node pkg="formation" type="robot_mavlink_bridge.py" name="robot_mavlink_bridge_$(arg id)"  required="true" output="screen">
        </node>
	</group>
	
</launch>
